tests/Integration/Behaviour/Features/Context/Domain/AttributeGroupFeatureContext.php

```diff
@@ -30,15 +30,118 @@
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Query\GetAttributeGroupForEditing;
 use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Query\GetAttributeGroupList;
 use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\QueryResult\AttributeGroup;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\QueryResult\EditableAttributeGroup;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\ValueObject\AttributeGroupId;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\ValueObject\AttributeGroupType;
 use PrestaShop\PrestaShop\Core\Domain\Product\AttributeGroup\Query\GetProductAttributeGroups;
 use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
 use RuntimeException;
+use Tests\Integration\Behaviour\Features\Context\SharedStorage;
+use Tests\Integration\Behaviour\Features\Context\Util\NoExceptionAlthoughExpectedException;
 use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
 
 class AttributeGroupFeatureContext extends AbstractDomainFeatureContext
 {
+    /**
+     * @When I create attribute group :reference with specified properties:
+     */
+    public function createAttributeGroup(string $reference, TableNode $node): void
+    {
+        $data = $this->localizeByRows($node);
+
+        $attributeGroupId = $this->createAttributeGroupUsingCommand($data['name'], $data['public_name'], $data['type']);
+
+        $this->getSharedStorage()->set($reference, $attributeGroupId->getValue());
+    }
+
+    /**
+     * @When I edit attribute group :reference with specified properties:
+     */
+    public function editAttributeGroup(string $reference, TableNode $node): void
+    {
+        $attributeGroupId = $this->referenceToId($reference);
+        $data = $this->localizeByRows($node);
+
+        $this->editAttributeGroupUsingCommand($attributeGroupId, $data['name'], $data['public_name'], $data['type']);
+    }
+
+    /**
+     * @When I delete attribute group :reference
+     */
+    public function deleteManufacturer(string $reference): void
+    {
+        $attributeGroupId = $this->referenceToId($reference);
+
+        $this->getCommandBus()->handle(new DeleteAttributeGroupCommand($attributeGroupId));
+    }
+
+    /**
+     * @Then attribute group :reference should be deleted
+     */
+    public function assertAttributeGroupIsDeleted(string $reference): void
+    {
+        $attributeGroupId = $this->referenceToId($reference);
+
+        try {
+            $this->getQueryBus()->handle(new GetAttributeGroupForEditing($attributeGroupId));
+
+            throw new NoExceptionAlthoughExpectedException(sprintf('Attribute group %s exists, but it was expected to be deleted', $reference));
+        } catch (AttributeGroupNotFoundException $e) {
+            $this->getSharedStorage()->clear($reference);
+        }
+    }
+
+    /**
+     * @Then attribute group :reference :field should be :value
+     */
+    public function assertFieldValue(string $reference, string $field, string $value): void
```

Comment:
I think now that you have the `assertAttributeGroupProperties` for assertion by table values you don't need this method anymore, you should also be able to remove `assertFieldWithLangValue`